functions/server/fn_startRecord.sqf

```diff
@@ -12,124 +14,106 @@ diag_log format ["grad replay: starting record with precision %1", _precision];
-						_color = [_color select 0, _color select 1, _color select 2, 0.5];
-						_groupname = "unconscious";
-					};
+			_side = side _unit;
+			_colorID = [_side] call GRAD_replay_fnc_getSideColorID;
+			_type = typeOf _veh;
+			_icon = getText (configfile >> "CfgVehicles" >> _type >> "icon");
 
-					
+			// mark funkwagen if he is sending in red // speciality for mission "breaking contact"
+			if (_type isEqualTo "rhs_gaz66_r142_vv" && {_veh getVariable ["tf_range",0] == 50000}) then {
+				_colorID = 11;
+			};
 
-					if (!alive _unit && _isMan) then {
-						_groupname = _unit getVariable ["GRAD_replay_persistentName", ""];
-						_color = [.2,.2,.2,1];
-					};
+			// speciality for mission "breaking contact"
+			if (_type isEqualTo "Land_DataTerminal_01_F" && !(isNil "GRAD_TERMINAL_ACTIVE") && {GRAD_TERMINAL_ACTIVE}) then {
+				_colorID = 11;
+			};
 
-					if (!alive _unit && _isEmptyVehicle) then {
-						_groupname = _unit getVariable ["GRAD_replay_persistentName", ""];
-						_color = [.2,.2,.2,1];
-					};
+			if (_isEmptyVehicle) then {
+				_colorID = [sideEmpty] call GRAD_replay_fnc_getSideColorID;
+			};
 
-					// current values: position, side, kindof
-					if ((_isMan && _side in GRAD_REPLAY_SIDES && (_unit getVariable ["grad_gcamspec_firstSpawn", true])) || 
-						_isEmptyVehicle ||
-						(_isCustomObject && _side in GRAD_REPLAY_SIDES ||
-						_isAI)) then {
-						[[_icon,_color,_pos,_dir,_name,_groupname]] call GRAD_replay_fnc_storeValue;
-					};
-				// };
+			_dir = round (getDir _veh);
 
-				// diag_log format ["grad replay: storing %1, %2, %3, %4, %5, %6", _unit,_color,_pos,_dir,_special,_veh];
+			if (_unit getVariable ["ACE_isUnconscious", false]) then {
+				_colorID = _colorID + 5;
+				_groupname = "unconscious";
+			};
 
-				
-			
-			} forEach _trackedUnits;
+			if (!alive _unit && (_isMan || _isEmptyVehicle)) then {
+				_groupname = _unit getVariable ["GRAD_replay_persistentName", ""];
+				_colorID = 10;
+			};
 
-			if (count GRAD_REPLAY_DATABASE_TEMP > 0) then {
+			[_currentUnitData,_nextTickData,_unitID,[_icon,_colorID,_pos,_dir,_name,_groupname]] call GRAD_replay_fnc_storeValue;
+		};
 
-				_hour = floor daytime;
-				_minute = floor ((daytime - _hour) * 60);
-				_second = floor (((((daytime) - (_hour))*60) - _minute)*60);
-				_emptyZeroMinute = "";
-				_emptyZeroSecond = "";
-				if (_minute < 10) then { _emptyZeroMinute = "0"; };
-				if (_second < 10) then { _emptyZeroSecond = "0"; };
-				_time24 = text format ["%1:%2%3:%4%5",_hour,_emptyZeroMinute,_minute,_emptyZeroSecond,_second];
-				GRAD_REPLAY_DATABASE_TEMP append [str _time24];
-
-		    	GRAD_REPLAY_DATABASE append [GRAD_REPLAY_DATABASE_TEMP];
-			};
-			GRAD_REPLAY_DATABASE_TEMP = [];
-		    
-		    // end recording and start playback
-		    if (GRAD_REPLAY_RECORDING_STOPPED) then {
-		    	[_handle] call CBA_fnc_removePerFrameHandler;
+		false
+	} count _trackedUnits;
```

Comment:
use `forEach` and get rid of the `false` at the end.

Reply:
Hold up, using `count` instead of `forEach` is commy level of scripting. Are you saying it doesn't make a difference?

Reply:
It does
`count` is faster than `forEach`.
But. If you have to add a additional call to the `false` command and make your overall loop slower with that.. It's worse.